### PR TITLE
HARROW-976 [Tencent Video Player] Implement Tencent super player videoxblock

### PR DESCRIPTION
**Pull request by UvgenGen**
[HARROW-976](https://youtrack.raccoongang.com/issue/HARROW-976) [Tencent Video Player] Implement Tencent super player videoxblock

### CODE REVIEW

File: video_xblock/backends/tencent.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
```
Comment (sendr) [inline video_xblock/backends/tencent.py:1]: do we really need it in python 3.8?

File: video_xblock/backends/tencent.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+"""
+Tencent Video player plugin.
```
Comment (sendr) [inline video_xblock/backends/tencent.py:3]:
```suggestion
Tencent Video player backend.
```

File: video_xblock/backends/tencent.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    def validate_data(self, validation, data):
+        """
+        Validate app id value which is mandatory.
+
+        Attributes:
+            validation (xblock.validation.Validation): Object containing validation information for an xblock instance.
+            data (xblock.internal.VideoXBlockWithMixins): Object containing data on xblock.
+        """
+        app_id_is_empty = data.app_id in ['', u'']  # pylint: disable=unsubscriptable-object
```
Comment (sendr) [inline video_xblock/backends/tencent.py:49]: let's remove `u''`

File: video_xblock/backends/tencent.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        Return a Fragment required to render video player on the client side.
+        """
+        context['player_state'] = json.dumps(context['player_state'])
+        context['lang_code'] = get_language()
+
+        frag = Fragment(
+            self.render_template('tencent.html', **context)
+        )
+        frag.add_javascript(
+            self.resource_string('static/js/base.js')
```
Comment (sendr) [inline video_xblock/backends/tencent.py:79]: please check whether we need it at all, thanks.

File: video_xblock/tests/unit/test_backends.py [hunk 5]
```diff
@@ -140,6 +143,10 @@ def test_get_player_html(self):
             'start_time', 'end_time', 'handout',
             'download_transcript_allowed', 'download_video_allowed',
         ],
+        [  # Tencent
+            'start_time', 'end_time', 'handout',
+            'download_transcript_allowed', 'download_video_allowed', 'download_video_url'
+        ],
```
Comment (sendr) [inline video_xblock/tests/unit/test_backends.py:149]: Let's please don't display advanced fields as we haven't implemented such functionality, thanks.

File: video_xblock/backends/tencent.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        app_id_is_empty = data.app_id in ['', u'']  # pylint: disable=unsubscriptable-object
+        # Validate provided app id
+        if app_id_is_empty:
+            # AppId field is mandatory
+            self.add_validation_message(
+                validation,
+                _(u"AppID can not be empty. Please provide a valid Tencent AppID.")
+            )
+
+    fields_help = {
```
Comment (sendr) [inline video_xblock/backends/tencent.py:58]: move it please to top

File: video_xblock/backends/tencent.py (hunk 4 above)
Comment (UvgenGen) [reply video_xblock/backends/tencent.py:79]: Checked, it's need.

File: video_xblock/tests/unit/test_backends.py [hunk 7]
```diff
@@ -140,6 +144,16 @@ def test_get_player_html(self):
             'start_time', 'end_time', 'handout',
             'download_transcript_allowed', 'download_video_allowed',
         ],
+        [],  # Tencent
+    ]
+
+    expected_advanced_tab = [
```
Comment (sendr) [inline video_xblock/tests/unit/test_backends.py:150]:
```suggestion
    expected_advanced_tab_enabled = [
```

File: video_xblock/tests/unit/test_backends.py [hunk 8]
```diff
@@ -152,6 +166,30 @@ def test_basic_advanced_fields(self, backend, expected_basic_fields, expected_ad
...
+        """
+        context = {}
+        test_js_context = {
+            'advancedTabEnabled': expected_advanced_tab,
+        }
+        player = self.player[backend](self.xblock)
+        player_mock.return_value = player
+        self.xblock.runtime.handler_url = Mock()
+        update_default_transcripts_mock.return_value = (
+            ['stub1', 'stub2'], 'Stub autoupload messate'
```
Comment (sendr) [inline video_xblock/tests/unit/test_backends.py:188]: typo?

**Review by sendr — APPROVED** [review]

**Pull request merged**